test/range/container/container_of_container_test.cpp

```diff
@@ -0,0 +1,345 @@
+{
+    EXPECT_TRUE(container_concept<TypeParam>);
+    EXPECT_TRUE(container_of_container_concept<TypeParam>);
+}
+
+TYPED_TEST(container_of_container, construction)
+{
+    TypeParam t1;
+    TypeParam t2{};
+//     EXPECT_EQ(t1, t2); // doesnt work because of conflict between ranges and gtest
+    EXPECT_TRUE(t1 == t2);
+
+    // initializer list
+    TypeParam t3{"ACGT"_dna4, "ACGT"_dna4, "GAGGA"_dna4};
+    TypeParam t4 = {"ACGT"_dna4, "ACGT"_dna4, "GAGGA"_dna4};
+    EXPECT_TRUE(t3 == t4);
+
+    // n * value
+    TypeParam t5{2, "ACGT"_dna4};
+    // from another TypeParam's sub-range
+    TypeParam t6{t3.begin(), t3.begin() + 2};
+    EXPECT_TRUE(t5 == t6);
+
+    std::vector<std::vector<dna4>> other_vector{"ACGT"_dna4, "ACGT"_dna4, "GAGGA"_dna4};
+    // direct from another container-of-container
+    TypeParam t7{other_vector};
+    // from another container-of-container's sub-range
+    TypeParam t8{other_vector.cbegin(), other_vector.cend()};
+    EXPECT_TRUE(t3 == t7);
+    EXPECT_TRUE(t7 == t8);
+}
+
+TYPED_TEST(container_of_container, assign)
+{
+    TypeParam t1{"ACGT"_dna4, "ACGT"_dna4, "GAGGA"_dna4};
+    TypeParam t2{"ACGT"_dna4, "ACGT"_dna4};
+    std::vector<std::vector<dna4>> other_vector{"ACGT"_dna4, "ACGT"_dna4, "GAGGA"_dna4};
+
+    // n * value
+    TypeParam t3;
+    t3.assign(2, "ACGT"_dna4);
+    EXPECT_TRUE(t3 == t2);
+
+    // from another container-of-container's sub-range
+    TypeParam t4;
+    t4.assign(other_vector.cbegin(), other_vector.cend());
+    EXPECT_TRUE(t4 == t1);
+
+    // initializer list
+    TypeParam t5, t6;
+    t5.assign({"ACGT"_dna4, "ACGT"_dna4, "GAGGA"_dna4});
+    t6 = {"ACGT"_dna4, "ACGT"_dna4, "GAGGA"_dna4};
+    EXPECT_TRUE(t5 == t1);
+    EXPECT_TRUE(t6 == t1);
+
+    // direct from another container-of-container
+    if constexpr (std::is_same_v<TypeParam, concatenated_sequences<std::vector<dna4>>>)
+    {
+        TypeParam t7, t8;
+        t7.assign(other_vector);
+        t8 = other_vector;
+        EXPECT_TRUE(t7 == t1);
+        EXPECT_TRUE(t8 == t1);
+    }
+}
+
+TYPED_TEST(container_of_container, iterators)
+{
+    TypeParam t1{"ACGT"_dna4, "ACGT"_dna4, "GAGGA"_dna4};
+    TypeParam const t2{"ACGT"_dna4, "ACGT"_dna4, "GAGGA"_dna4};
+
+    // begin
+    EXPECT_TRUE(dna4_vector(*t1.begin())  == "ACGT"_dna4);
+    EXPECT_TRUE(dna4_vector(*t1.cbegin()) == "ACGT"_dna4);
+    EXPECT_TRUE(dna4_vector(*t2.begin())  == "ACGT"_dna4);
+    EXPECT_TRUE(dna4_vector(*t2.cbegin()) == "ACGT"_dna4);
+
+    // end and arithmetic
+    EXPECT_TRUE(dna4_vector(*(t1.end()  - 1)) == "GAGGA"_dna4);
+    EXPECT_TRUE(dna4_vector(*(t1.cend() - 1)) == "GAGGA"_dna4);
+    EXPECT_TRUE(dna4_vector(*(t2.end()  - 1)) == "GAGGA"_dna4);
+    EXPECT_TRUE(dna4_vector(*(t2.cend() - 1)) == "GAGGA"_dna4);
+
+    // convertibility between const and non-const
+    EXPECT_TRUE(t1.cend() == t1.end());
+
+    // writability
+    (*t1.begin())[0] = dna4::T;
+    EXPECT_TRUE(dna4_vector(*t1.begin())  == "TCGT"_dna4);
```

Comment:
Because gtest compares via stream and the views get streamed to `[T,C,G,T]` while the dna4vector results in `TCGT`.